cmd/sovereignnode/chainSimulator/chainSimulator.go

```diff
@@ -0,0 +1,99 @@
+	"github.com/multiversx/mx-chain-go/sovereignnode/incomingHeader"
+	sovRunType "github.com/multiversx/mx-chain-go/sovereignnode/runType"
+
+	"github.com/multiversx/mx-chain-core-go/core"
+)
+
+// ArgsSovereignChainSimulator holds the arguments for sovereign chain simulator
+type ArgsSovereignChainSimulator struct {
+	SovereignExtraConfig config.SovereignConfig
+	ChainSimulatorArgs   chainSimulator.ArgsChainSimulator
+}
+
+// NewSovereignChainSimulator will create a new instance of sovereign chain simulator
+func NewSovereignChainSimulator(args ArgsSovereignChainSimulator) (chainSimulatorIntegrationTests.ChainSimulator, error) {
+	args.ChainSimulatorArgs.CreateGenesisNodesSetup = func(nodesFilePath string, addressPubkeyConverter core.PubkeyConverter, validatorPubkeyConverter core.PubkeyConverter, _ uint32) (sharding.GenesisNodesSetupHandler, error) {
+		return sharding.NewSovereignNodesSetup(&sharding.SovereignNodesSetupArgs{
+			NodesFilePath:            nodesFilePath,
+			AddressPubKeyConverter:   addressPubkeyConverter,
+			ValidatorPubKeyConverter: validatorPubkeyConverter,
+		})
+	}
+	args.ChainSimulatorArgs.CreateRatingsData = func(arg rating.RatingsDataArg) (process.RatingsInfoHandler, error) {
+		return rating.NewSovereignRatingsData(arg)
+	}
+	args.ChainSimulatorArgs.CreateIncomingHeaderSubscriber = func(config *config.NotifierConfig, dataPool dataRetriever.PoolsHolder, mainChainNotarizationStartRound uint64, runTypeComponents factory.RunTypeComponentsHolder) (process.IncomingHeaderSubscriber, error) {
+		return incomingHeader.CreateIncomingHeaderProcessor(config, dataPool, mainChainNotarizationStartRound, runTypeComponents)
+	}
+	args.ChainSimulatorArgs.CreateRunTypeComponents = func(argsRunType runType.ArgsRunTypeComponents) (factory.RunTypeComponentsHolder, error) {
+		return createSovereignRunTypeComponents(argsRunType, args.SovereignExtraConfig)
+	}
+	args.ChainSimulatorArgs.NodeFactory = node.NewSovereignNodeFactory()
+	args.ChainSimulatorArgs.ChainProcessorFactory = NewSovereignChainHandlerFactory()
+
+	return chainSimulator.NewChainSimulator(args.ChainSimulatorArgs)
+}
+
+// LoadSovereignConfigs loads sovereign configs
+func LoadSovereignConfigs(configsPath string) (*config.EpochConfig, *config.EconomicsConfig, *config.SovereignConfig, error) {
+	epochConfig, err := common.LoadEpochConfig(path.Join(configsPath, "enableEpochs.toml"))
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	economicsConfig, err := common.LoadEconomicsConfig(path.Join(configsPath, "economics.toml"))
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	sovereignExtraConfig, err := sovereignConfig.LoadSovereignGeneralConfig(path.Join(configsPath, "sovereignConfig.toml"))
+	sovereignExtraConfig.OutGoingBridgeCertificate = config.OutGoingBridgeCertificate{
+		CertificatePath:   "/home/ubuntu/MultiversX/testnet/config/certificate.crt",
+		CertificatePkPath: "/home/ubuntu/MultiversX/testnet/config/private_key.pem",
```

Comment:
Please remove this !!! 
These files only exist on your machine.
We don't even need these, we don't create a grpc connection.